### DISCUSSION TITLE
Terminology for delimiter inclusion

### DISCUSSION THREAD

**Discussion by pokey** [cursorless-dev/cursorless#327] [General]
As part of implementing https://github.com/pokey/cursorless-vscode/pull/324, we'll be formalizing identifiers to indicate whether to include delimiters, exclude delimiters, or select delimiters themselves.  Currently the terms are `excludeDelimiters`, `includeDelimiters`, and `delimitersOnly`.  However, we'd like to use this same machinery to allow things like unwrapping / rewrapping `if` statements, so I'm not sure "delimiters" is the term we want to use to refer to the guard of an `if` statement.  What about the following rename:

- `includeDelimiters` => `fullScope`,
- `excludeDelimiters` => `interiorOnly`, and
- `delimitersOnly` => `excludeInterior`?

I believe that these terms work for both delimiter pairs, as well as things like strings and `if` statements.

**Comment by AndreasArvidsson** [comment]
Works for me

**Comment by soheeyang** [comment]
Sounds good to me, too!

**Comment by pokey** [comment]
Also, given that in the future these will become standalone modifiers, `fullScope` will just be a no-op, so I think we could just drop it.  I have an idea for how to make that work so that the grammar won't have to change
